docs/sphinx/examples/serialization.rst

```diff
@@ -0,0 +1,146 @@
+
+To compile this program, go to your |hpx| build directory (see
+:ref:`hpx_build_system` for information on configuring and building |hpx|) and
+enter:
+
+.. code-block:: shell-session
+
+   $ make examples.quickstart.custom_serialization
+
+To run the program type:
+
+.. code-block:: shell-session
+
+   $ ./bin/custom_serialization
+
+This should print:
+
+.. code-block:: text
+
+   Rectangle(Point(x=0,y=0),Point(x=0,y=5))
+   gravity.g = 9.81%
+
+
+Serialization Requirements
+===========================
+In order to serialize objects in HPX, at least one of the following criteria must be met:
+
+In the case of default constructible objects:
+
+* The object is an empty type.
+* Has a serialization function as shown in this tutorial.
+* All members are accessible publicly and they can be used in structured binding contexts.
+
+Otherwise:
+
+* They need to have  :ref:`special serialization support <Serializing non default constructable classes>`.
+
+Member function serialization
+-----------------------------
+
+.. literalinclude:: ../../examples/quickstart/custom_serialization.cpp
+   :language: c++
+   :start-after: //[PointMemberSerialization
+   :end-before: //]
+
+Notice that ``PointMemberSerialization`` is defined as bitwise serializable
+(see :ref:`bitwise_serialization` for more details).
+HPX is also able to recursively serialize composite classes and structs
+given that its members are serializable.
+
+.. literalinclude:: ../../examples/quickstart/custom_serialization.cpp
+   :language: c++
+   :start-after: //[RectangleMemberSerialization
+   :end-before: //]
+
+
+
+Free function serialization
+-----------------------------
+
+In order to decouple your models from HPX, HPX also allows for the definition
+of free function serializers.
+
+.. literalinclude:: ../../examples/quickstart/custom_serialization.cpp
+   :language: c++
+   :start-after: //[RectangleFREE
+   :end-before: //]
+
+Even if you can't modify a class to befriend it, you can still be able to
+serialize your class provided that your class is default constructable
+and you are able to reconstruct it yourself.
+
+.. literalinclude:: ../../examples/quickstart/custom_serialization.cpp
+   :language: c++
+   :start-after: //[PointClass
+   :end-before: //]
+
+
+Serializing non default constructable classes
+----------------------------------------------
+.. _non-default_constructable
+
+Some classes don't provide any default constructor.
+
+.. literalinclude:: ../../examples/quickstart/custom_serialization.cpp
+   :language: c++
+   :start-after: //[PlanetWeightCalculator
+   :end-before: //]
+
+In this case you have to define a ``save_construct_data`` and ``load_construct_data`` in which you
+do the serialization yourself.
+
+.. literalinclude:: ../../examples/quickstart/custom_serialization.cpp
+   :language: c++
+   :start-after: //[save_construct_data
+   :end-before: //]
+
+.. _bitwise_serialization:
+
+Bitwise serialization for arithmatic data
```

Comment:
```suggestion
Bitwise serialization for arithmetic data
```

Reply:
This applies to all bitwise copyable types, BTW.

Reply:
I changed the title to `Bitwise serialization for bitwise copyable data` to be more explicit about its benefits.